### ISSUE TITLE
[Question]: how would you run debugpy?

### ISSUE THREAD

**Issue by matan-h** [chaquo/chaquopy#1050]
VSCode [debug](https://code.visualstudio.com/docs/python/debugging) Python using their open-source debugger called [debugpy](https://github.com/microsoft/debugpy).
This debugger [contains](https://github.com/microsoft/debugpy/tree/main/src/debugpy/_vendored/pydevd) another debugger called [pydev](https://github.com/fabioz/Pydev).

Pydev [as a standalone](https://pypi.org/project/pydevd-pycharm/) runs well under Chaquopy.
However, debugpy includes its version of pydev in this path:
`src/debugpy/_vendored/pydevd`
so the "domain" path would be `debugpy._vendored.pydevd`.
The init file (of `debugpy._vendored.pydevd`) has this code:
```python
try:
    # Just empty packages to check if they're in the PYTHONPATH.
    import _pydev_bundle
except ImportError:
    # On the first import of a pydevd module, add pydevd itself to the PYTHONPATH
    # if its dependencies cannot be imported.
    sys.path.append(os.path.dirname(os.path.abspath(__file__)))
    import _pydev_bundle
```
So the code tries to import `_pydev_bundle` and appends the dirname of the current file to `sys.path`.

But here is the problem:
1. If I just try to run `debugpy` without any hooks, it would throw this Java/Python error which I don't understand: (probably something went wrong with the way they used `importlib`)
```java
E/AndroidRuntime: java.lang.RuntimeException: Unable to start activity ComponentInfo{com.matan_h.beedebug.bee_debug/org.beeware.android.MainActivity}: com.chaquo.python.PyException: AttributeError: 'AssetFinder' object has no attribute 'extract_packages'
E/AndroidRuntime:       at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3782)
...
E/AndroidRuntime: Caused by: com.chaquo.python.PyException: AttributeError: 'AssetFinder' object has no attribute 'extract_packages'
E/AndroidRuntime:       at <python>.java.android.importer.extract_dir(importer.py:476)
E/AndroidRuntime:       at <python>.java.android.importer.find_module(importer.py:449)
...
E/AndroidRuntime:       at <python>.importlib.import_module(__init__.py:127)
E/AndroidRuntime:       at <python>.debugpy._vendored.force_pydevd.<module>(force_pydevd.py:34)
```
2. After I created a hook that makes `importlib.import_module` equal to `__import__`, I got the expected error:
```python
ModuleNotFoundError: No module named '_pydev_bundle'
```
Probably something in the way `sys.path` works in a frozen context.
And since `_pydev_bundle` is a directory, I don't know how to create a redirect (or any other way).

Does anyone have any idea (or hacky workaround) on how to run `debugpy`?

**Comment by mhsmith** [comment]
> AttributeError: 'AssetFinder' object has no attribute 'extract_packages'

This is probably the same as #820, whose fix will be released in Chaquopy 15.0.1 in a few days.

The "debugger" directory in the root of the repository contains some notes that you might find useful, though they're several years old so I don't know how relevant they still are.

**Comment by matan-h** [comment]
Thanks, the issue you referred (#820) is indeed look like the `AssetFinder` error.

About the "/debugger" directory: it's about running the [pydevd of pycharm](https://pypi.org/project/pydevd-pycharm/), which is nowadays is seemed to be working fine (the pydevd code is running and connect to my netcat, but I can't test this as I don't have pycharm professional and I don't know how to run any other pydevd server).

However, maybe I should modify the `debugpy` project as you did to the `pydevd` some years ago.
